src/framework/entity.js

```diff
@@ -601,6 +604,24 @@ class Entity extends GraphNode {
         return clone;
     }
 
+    _getSortedComponents() {
+        _sortedArray.length = 0;
+
+        let needSort = false;
+        for (const component of Object.values(this.c)) {
```

Comment:
use this loop as before, to avoid the array allocation inside `Object.values`:
```
        for (const type in components) {
            if (components.hasOwnProperty(type)) {
```